dapps/marketplace/src/hoc/withCurrencyBalances.js

```diff
@@ -30,12 +30,15 @@ function withCurrencyBalances(WrappedComponent) {
       variables={{ account: props.wallet, tokens: props.targets }}
       fetchPolicy="network-only"
     >
-      {({ data }) => (
-        <WrappedComponent
-          {...props}
-          currencies={get(data, 'currencies') || []}
-        />
-      )}
+      {({ data, loading }) => {
+        if (loading) return null
```

Comment:
Hrm in general I think we should avoid returning null from higher order components as it could cause items to not be rendered at all instead of showing a loading state or similar further down the chain